### ISSUE TITLE
Support for `eval` in functional_call

### ISSUE THREAD

**Issue by Silur** [pytorch/pytorch#104768]
### 🚀 The feature, motivation and pitch

`functional_call` on a model currently runs in `.train()` mode by default, which will fail on models containing for eg `BatchNorm2d` layers which explicitly need the `training` variable to be `False` at inference.
`functional_call` currently has no way of calling layers as such, hence fails on BatchNorm2d layers.

**Comment by jbschlosser** [comment]
> `functional_call` on a model currently runs in `.train()` mode by default

Hey @Silur, I believe `functional_call` runs the model on whatever mode it's currently in. For example:

```python
m = torch.nn.Dropout(p=0.5)
m.eval()
x = torch.randn(3, 4)
output = functional_call(m, {}, x)
```

No dropout will be applied on the output if `m.eval()` is called beforehand. Does this match what you're seeing?

**Comment by albanD** [comment]
Why not just do `model.eval()` before the functional_call?
